zebedee-cms/src/main/java/com/github/onsdigital/zebedee/util/SlackNotification.java

```diff
@@ -27,88 +43,163 @@
+    private static Cache<String, AbstractMap.SimpleEntry<String, String>> collectionSlackMessageTimestamps = CacheBuilder.newBuilder()
+            .maximumSize(1000)
+            .expireAfterWrite(10, TimeUnit.MINUTES)
+            .build();
+
+    public static void alarm(String alarm, PostMessageField... args) {
+        PostMessage pm = new PostMessage(slackUsername, slackAlarmChannel, PostMessage.Emoji.HeavyExclamationMark);
+        PostMessageAttachment attch = new PostMessageAttachment(alarm, "Alarm", PostMessageAttachment.Color.Danger);
+        pm.attachments.add(attch);
+        attch.fields.addAll(Arrays.asList(args));
+        postSlackMessage(pm, null, true);
+    }
 
-        if (c.getDescription().approvalStatus != null) {
-            msg.append("\nApproval status: " + c.getDescription().approvalStatus.toString());
+    private static void addCollectionAttachment(PostMessageAttachment attch, Collection collection) {
+        attch.fields.add(new PostMessageField("Collection", "<" + Configuration.getFlorenceUrl() + "/florence/collections/" + collection.getDescription().getId() + "|" + collection.getDescription().getName()+ ">", true));
+
+        if(collection.getDescription().getPublishDate() == null) {
+            attch.fields.add(new PostMessageField("Publish date", "Manual", true));
+        } else {
+            attch.fields.add(new PostMessageField("Publish date", slackFieldFormatVague.format(collection.getDescription().getPublishDate()), true));
         }
 
-        if (c.getDescription().publishTransactionIds != null
-                && !c.getDescription().publishTransactionIds.isEmpty()) {
+        if (collection.getDescription().approvalStatus != null) {
+            attch.fields.add(new PostMessageField("Approval status", collection.getDescription().approvalStatus.toString(), true));
+        }
 
-            msg.append("\nTransaction IDs:");
-            c.getDescription().publishTransactionIds.entrySet()
-                    .stream().forEach(e -> msg.append("\n - " + e.getValue()));
+        if (collection.getDescription().publishTransactionIds != null
+                && !collection.getDescription().publishTransactionIds.isEmpty()) {
+            attch.fields.add(new PostMessageField("Transaction IDs", StringUtils.join(collection.getDescription().publishTransactionIds.values(), "\n"), true));
         }
-        alarm(msg.toString());
     }
 
-    public static void alarm(String message) {
-        String slackUsername = "Alarm";
-        String slackEmoji = ":heavy_exclamation_mark:";
-        send(message, slackToken, slackAlarmChannel, slackUsername, slackEmoji);
-    }
+    private static void addCollectionAttachment(PostMessageAttachment attch, CollectionBase collection) {
+        attch.fields.add(new PostMessageField("Collection", "<" + Configuration.getFlorenceUrl() + "/florence/collections/" + collection.getId() + "|" + collection.getName() + ">", true));
```

Comment:
Appending these values inline feels bit nasty, IMO it would be better to have a format method and call that instead. it also makes reading this method easier as the input is shorter.